proxy/src/transport/tls/config.rs

```diff
@@ -258,3 +279,100 @@ fn set_common_settings(versions: &mut Vec<rustls::ProtocolVersion>) {
     // Only enable TLS 1.2 until TLS 1.3 is stable.
     *versions = vec![rustls::ProtocolVersion::TLSv1_2]
 }
+
+
+#[cfg(test)]
+mod tests {
+    use tls::{CommonSettings, Identity, ServerConfig};
+    use super::{CommonConfig, Error};
+    use config::Namespaces;
+    use std::path::PathBuf;
+
+    struct Strings {
+        pod_name: &'static str,
+        pod_ns: &'static str,
+        controller_ns: &'static str,
+        trust_anchors: &'static str,
+        end_entity_cert: &'static str,
+        private_key: &'static str,
+    }
+
+    fn settings(s: &Strings) -> CommonSettings {
+        let dir = PathBuf::from("src/transport/tls/testdata");
+        let namespaces = Namespaces {
+            pod: s.pod_ns.into(),
+            tls_controller: Some(s.controller_ns.into()),
+        };
+        let service_identity = Identity::try_from_pod_name(&namespaces, s.pod_name).unwrap();
+        CommonSettings {
+            trust_anchors: dir.join(s.trust_anchors),
+            end_entity_cert: dir.join(s.end_entity_cert),
+            private_key: dir.join(s.private_key),
+            service_identity,
+        }
+    }
+
+    #[test]
+    fn can_construct_server_config_from_valid_settings() {
+        let settings = settings(&Strings {
+            pod_name: "foo",
+            pod_ns: "ns1",
+            controller_ns: "conduit",
+            trust_anchors: "ca1.pem",
+            end_entity_cert: "foo-ns1-ca1.crt",
+            private_key: "foo-ns1-ca1.p8",
+        });
+        let config = CommonConfig::load_from_disk(&settings).unwrap();
+        let _: ServerConfig = ServerConfig::from(&config); // Infallible.
+    }
+
+    #[test]
+    fn recognize_ca_did_not_issue_cert() {
+        let settings = settings(&Strings {
+            pod_name: "foo",
+            pod_ns: "ns1",
+            controller_ns: "conduit",
+            trust_anchors: "ca2.pem", // Mismatch
+            end_entity_cert: "foo-ns1-ca1.crt",
+            private_key: "foo-ns1-ca1.p8",
+        });
+        match CommonConfig::load_from_disk(&settings) {
+            Err(Error::EndEntityCertIsNotValid(_)) => (),
+            _ => unreachable!()
```

Comment:
Nit: Is `unreachable!()` the best way to fail these tests? It seems like panicking with a message would lead to clearer error messages if someone introduces a regression.

Reply:
I rewrote these to replace the last line of each with `r => unreachable!("CommonConfig::load_from_disk returned {:?}", r),`. I think this is the best that can be done right now because some error types don't implement `PartialEq`.

Reply:
Yeah, that seems good for now, thanks for making that change.